project.clj

```diff
@@ -10,14 +10,23 @@
   :profiles {:dev {:dependencies [[org.clojure/clojure "1.11.1"]
                                   [junit/junit "4.13"]]}
              :test {:dependencies [[junit/junit "4.13"]]}}
-  :java-source-paths ["src" "test/java"]
+  :java-source-paths ["src"]
+  :test-paths ["test/clojure"]
   :javac-options ["-target" "1.8" "-source" "1.8"]
-  :plugins [[lein-junit "1.1.9"]]
   :junit ["test/java"]
 
   ;; Maven properties for the Maven God
   :scm {:url "git@github.com:clj-commons/dirigiste.git"}
-  :pom-addition ([:organization
+  :pom-plugins [[org.codehaus.mojo/build-helper-maven-plugin "1.7"
+                {:executions
+                 ([:execution [:id "add-test-source"]
+                              [:phase "generate-test-sources"]
+                              [:goals [:goal "add-test-source"]]
+                              [:configuration [:sources [:source "test/java"]]]])}]]
```

Comment:
This is a bit weird, since this is also a built-in plugin that `lein pom` automatically adds. Adding it again makes maven complain. Hopefully this isn't important.

Reply:
Yes, if only `java-test-path` was supported. [1]

[1] : https://github.com/technomancy/leiningen/blob/github/src/leiningen/pom.clj#L251-L253